tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Control/AsyncType.fs

```diff
@@ -167,17 +172,118 @@ type AsyncType() =
         try
             let result = t.Wait(300)
             Assert.IsFalse (result)
-        with :? AggregateException -> Assert.Fail "Task should not finish, jet"
+        with :? AggregateException -> Assert.Fail "Task should not finish, yet"
 
         tcs.SetCanceled()
         
         try
             this.WaitASec t
-        with :? AggregateException as a ->
-            match a.InnerException with
-            | :? TaskCanceledException as t -> ()
+        with :? TaskCanceledException -> ()
+        Assert.IsTrue (t.IsCompleted, "Task is not completed")
+        Assert.IsTrue (t.IsCanceled, "Task is not cancelled")
+
+    [<Test>]
+    member this.StartAsTaskCancellationViaException () =
+        let cts = new CancellationTokenSource()
+        let tcs = TaskCompletionSource<unit>()
+        let a = async {
+            cts.CancelAfter (100)
+            do! tcs.Task |> Async.AwaitTask }
+#if FSCORE_PORTABLE_NEW || coreclr
```

Comment:
Does this still exist?

Reply:
cc @KevinRansom

Reply:
I'm taking a look through these various old flags now, will send a PR to flush this